Source/CarthageKit/Xcode.swift

```diff
@@ -1181,6 +1181,67 @@ public func parseSecuritySigningIdentities(securityIdentities securityIdentities
 		.ignoreNil()
 }
 
+public func projectsInDirectory(directoryURL: NSURL, configuration: String, platforms: Set<Platform> = []) -> SignalProducer<(ProjectLocator, [String]), CarthageError> {
+	let directoryLength = directoryURL.pathComponents!.count
+	return locateProjectsInDirectory(directoryURL)
+		.filter { project in
+			let fileURL = project.fileURL
+			let fileComponentsLength = fileURL.pathComponents!.count
+			return fileComponentsLength - directoryLength < 3 // within 2 subdirectories. Because we actually have a test for this.
```

Comment:
Apparently I'm dyslexic. I was going to point to https://github.com/Carthage/Carthage/blob/master/Source/CarthageKitTests/XcodeSpec.swift#L291-L298 and say that that's supposed to be testing that it's a subsubdirectory or earlier. My bad.

The tests earlier failed because the ReactiveCocoaLayout test fixture has an old version of Quick (0.6.0) committed to it's Carthage subfolder. Maybe it would be better to check if the project does not have a parent (or further up) "Carthage" folder? In either case, should probably remove that comment.